pkg/apis/testing/v1alpha1/testsuite_types.go

```diff
@@ -0,0 +1,138 @@
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// TestSuiteSpec defines the desired state of TestSuite
+type TestSuiteSpec struct {
+	// How many tests we want to execute at the same time.
+	// Depends on cluster size and it's load
+	ConcurrencyLevel int64 `json:"concurrencyLevel,omitempty"`
+	// You can specify test names explicitly.
+	TestNamesSelector []NamespacedTest `json:"testNamesSelector,omitempty"`
+	// Will run every test that depends of any of the component listed here
+	ComponentsSelector []string `json:"componentsSelector,omitempty"`
+	// Running all tests from suite cannot take more time that specified here
+	SuiteTimeout *metav1.Duration `json:"suiteTimeout,omitempty"`
+	// If specific TestDefinition does not define timeout, use this one
+	DefaultTestTimeout *metav1.Duration `json:"defaultTestTimeout,omitempty"`
```

Comment:
can we defaulting the value in TestDefinition? it will simplify our code, in reconcile u will have simple logic.


With this approach you will end up with sth like:
```go
var timeout time.Duration
if TestDefinition.spec.testTimeout != nil && !TestDefinition.spec.testTimeout.isZero() {
	timeout = TestDefinition.spec.testTimeout
} else {
	timeout = TestSuite.Spec.DefaultTestTimeout
}
```


but it can be simplified only if the default timeout is always present (so no timeout will not be supported)

Reply:
Then you would make TestDefinition.spec.Timeout required field, right?

Reply:
I am wondering if we can provide a reasonable default for a test timeout. Remember that this library is going to conquer the world. For Kyma we can propose some default, but I am not sure if it will be proper for other projects. 
Currently, we have only one default value: timeout for the whole TestSuite is set 1h.
If this default is incorrect, people have to modify their Testsuites, and usually, a number of Test Suites is significantly smaller than the number of TestDefinitions. If we use wrong default for test definition, then changes needs to be done in more places.

Reply:
I'd say we run it infinitely 😈